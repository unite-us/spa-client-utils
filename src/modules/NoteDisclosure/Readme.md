```jsx
  <NoteDisclosure className="classname-add-on" />
```
