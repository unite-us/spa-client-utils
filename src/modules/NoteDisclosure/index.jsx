import React from 'react';
import PropTypes from 'prop-types';
import InfoPanel from 'modules/InfoPanel/InfoPanel';

// eslint-disable-next-line max-len
const NOTE_DISCLOSURE_MESSAGE = 'Only include personally identifiable information (PII), protected health information (PHI), or other sensitive information if it is necessary to provide services to the client.';

const NoteDisclosure = ({ className }) => <InfoPanel className={className} message={NOTE_DISCLOSURE_MESSAGE} />;

NoteDisclosure.propTypes = {
  /* Class name */
  className: PropTypes.string,
};

NoteDisclosure.defaultProps = {
  className: '',
};

export default NoteDisclosure;
