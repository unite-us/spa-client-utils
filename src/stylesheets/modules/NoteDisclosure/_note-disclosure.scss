.note-disclosure {
  border-radius: 10px;
  display: flex;
  align-items: flex-start;
  background-color: $cloud-blue-color;
  padding: 20px;
  padding-left: 5px;

  &__icon {
    flex: 0 0 5%;
  }

  &__text {
    flex: 0 0 95%;
  }
}

@media only screen and (max-width: 1200px) {
  .note-disclosure {
    &__icon {
      flex: 0 0 10%;
    }
    
    &__text {
      flex: 0 0 90%;
    }
  }
}
