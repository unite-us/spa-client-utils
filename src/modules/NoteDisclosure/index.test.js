import React from 'react';
import renderer from 'react-test-renderer';
import NoteDisclosure from './';

describe('NoteDisclosure', () => {
  it('renders', () => {
    const props = {
      className: 'className-prop',
    };

    const comp = renderer.create(<NoteDisclosure {...props} />).toJSON();
    expect(comp).toMatchSnapshot();
  });
});
