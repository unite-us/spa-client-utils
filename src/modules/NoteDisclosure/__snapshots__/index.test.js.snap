// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteDisclosure renders 1`] = `
<div
  className="info-panel className-prop"
>
  <svg
    className="ui-icon info-panel__icon"
    height="16px"
    id=""
    style={
      Object {
        "display": "inline-block",
        "fill": "",
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 27 32"
    width="16px"
  >
    <path
      d="M13.714 2.286q3.732 0 6.884 1.839t4.991 4.991 1.839 6.884-1.839 6.884-4.991 4.991-6.884 1.839-6.884-1.839-4.991-4.991-1.839-6.884 1.839-6.884 4.991-4.991 6.884-1.839zM16 24.554v-3.393q0-0.25-0.161-0.42t-0.393-0.17h-3.429q-0.232 0-0.411 0.179t-0.179 0.411v3.393q0 0.232 0.179 0.411t0.411 0.179h3.429q0.232 0 0.393-0.17t0.161-0.42zM15.964 18.411l0.321-11.089q0-0.214-0.179-0.321-0.179-0.143-0.429-0.143h-3.929q-0.25 0-0.429 0.143-0.179 0.107-0.179 0.321l0.304 11.089q0 0.179 0.179 0.313t0.429 0.134h3.304q0.25 0 0.42-0.134t0.188-0.313z"
    />
  </svg>
  <p
    className="info-panel__text"
  >
    Only include personally identifiable information (PII), protected health information (PHI), or other sensitive information if it is necessary to provide services to the client.
  </p>
</div>
`;
